### ISSUE TITLE
500 errors happen

### ISSUE THREAD

**Issue by Phyks** [ampache/ampache#1231]
# Environment
 
 * Ampache version: latest dev
 * Web server + version: PHP5-FPM
 * Server operating system: Linux
 * Client operating system: Linux
 * Client type:
   - [x] Web
   - [ ] Ampache
   - [ ] SubSonic
   - [ ] Plex
   - [ ] DAAP
   - [ ] UPnP
   - [ ] WebDAV
 * Client application + version: Chromium, latest

# Description

In Chromium dev console, I got:
```
https://AMPACHE/server/ajax.server.php?page=index&action=similar_now_playing&media_id=77281&media_artist=424 Failed to load resource: the server responded with a status of 500 (Internal Server Error)

https://AMPACHE/waveform.php?song_id=75896 Failed to load resource: the server responded with a status of 504 (OK)

https://AMPACHE/waveform.php?song_id=82655 Failed to load resource: the server responded with a status of 504 (OK)
```

# Logs

From the webserver:
```
IP - - [30/May/2016:10:41:13 +0200] "POST /server/ajax.server.php?page=index&action=similar_now_playing&media_id=77281&media_artist=424 HTTP/1.1" 500 0              
"https://AMPACHE/index.php" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/47.0.2526.80 Safari/537.36"                                                       
2016/05/30 10:41:13 [error] 28828#0: *6069 FastCGI sent in stderr: "PHP message: PHP Fatal error:  Uncaught exception 'Requests_Exception' with message 'cURL error 28: Operation 
timed out after 10000 milliseconds with 0 bytes received' in /AMPACHE/lib/vendor/rmccue/requests/library/Requests/Transport/cURL.php:280                
Stack trace:       
#0 /AMPACHE/lib/vendor/rmccue/requests/library/Requests/Transport/cURL.php(123): Requests_Transport_cURL->process_response(false, Array)                
#1 /AMPACHE/lib/vendor/rmccue/requests/library/Requests.php(317): Requests_Transport_cURL->request('http://ws.audio...', Array, NULL, Array)            
#2 /AMPACHE/lib/vendor/rmccue/requests/library/Requests.php(194): Requests::request('http://ws.audio...', Array, NULL, 'GET', Array)                    
#3 /AMPACHE/lib/class/recommendation.class.php(51): Requests::get('http://ws.audio...', Array, Array)                                                   
#4 /AMPACHE/lib/class/recommendation.class.php(44): Recommendation::query_lastfm('http://ws" while reading response header from upstream, client: IP, server: AMPACHE, request: "POST /server/ajax.server.php?page=index&action=similar_now_playing&media_id=77281&media_artist=424 HTTP/1.1", upstream: "fastcgi://unix:/var/run/php5-fpm-b6.sock:", host: "AMPACHE", referrer: "https://AMPACHE/index.php"
```

I think these logs correspond to the first 500 error. I'm not sure about what is happening with the 504.

Thanks

cc @Niols

**Comment by Afterster** [comment]
Looks like a communication issue with ws.audioscrobbler API when requesting similar artists/songs. Maybe their API changed, should be checked (https://github.com/ampache/ampache/blob/6484b254e1446372fdc2cb74fad008b4cdba220f/lib/class/recommendation.class.php#L41).

**Comment by Phyks** [comment]
Ok, I'll check it and see if I can find something.

**Comment by Phyks** [comment]
I checked it quickly and I could not manage to do a `track.getsimilar` call manually, but I must have missed something. The only response I have is
```
<lfm status="ok"/>
```

Still, it should not matter as there is a response, so this should not raise the above mentioned exception.

Same thing, [in the doc](http://www.last.fm/api/show/track.getSimilar), I noticed that when doing a "track" query (with song title rather than mbid), one should provide both a `track` and an `artist`, which is not what is done [here](https://github.com/ampache/ampache/blob/6484b254e1446372fdc2cb74fad008b4cdba220f/lib/class/recommendation.class.php#L133).

Anyways, this should not be the cause of this issue and I am trying to find a reproducible way to generate such errors to investigate them further.
